pkg/controller/topology.go

```diff
@@ -562,6 +562,10 @@ func (t topologyTerm) less(other topologyTerm) bool {
 	return t.hash() < other.hash()
 }
 
+func (t topologyTerm) subset(other topologyTerm) bool {
+	return strings.Contains(other.hash(), t.hash())
```

Comment:
What if `rackA` and `rackAB` both exist?

What if the two hashes are `...#rackA,...#rackC` and `...#rackA,...#rackB,...#rackC`?

Reply:
good point. I should probably rely on checking individual key value pair. Thanks. Will fix this. I will also update the test case for this.

Reply:
Was a test case added for this?

Reply:
@msau42, Yes, I have modified the test case to cover this.

Reply:
I'm not seeing the case where an implementation using "Contains" would have incorrectly allowed "rackA" when the allowedTopologies was "rackAB".

Reply:
It is the other way around @msau42, it will not allow which should be allowed in few cases. The problem is since in the hash we sort all the key value pair and generate one string, so contains logic may fail when becasue of sorting, something comes in between even though all the key values are present but not contiguously. Check the unit test case, I have modified it to cover this case.  With the conatins logic that will fail.

Storageclass
allowedTopologies:
- matchLabelExpressions:
  - key: com.example.csi/disk
    values:
      - ssd
  - key: com.example.csi/zone
    values:
      - zone1

and nodes are like this:- 

node1: {"com.example.csi/zone": "zone1", "com.example.csi/rack": "rack1", "com.example.csi/disk": "ssd"}

node2: {"com.example.csi/zone": "zone2", "com.example.csi/rack": "rack2", "com.example.csi/disk": "ssd"}

node3: {"com.example.csi/zone": "zone2", "com.example.csi/rack": "rack1", "com.example.csi/disk": "nvme"}


As per storage class node1 should qualify that but the two hashes will be like :
storageclass :  com.example.csi/disk#ssd,com.example.csi/zone#zone1
node1:- com.example.csi/disk#ssd,com.example.csi/rack#rack1,com.example.csi/zone#zone1

Since rack is coming in between, so contains will work here even though it qualifies the storageclass requirement.

Reply:
I'm looking for a test case where:

allowedTopologies is something like:
```
					MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
						{
							Key:    "com.example.csi/disk",
							Values: []string{"ssd"},
						},
						{
							Key:    "com.example.csi/zone",
							Values: []string{"zone1"},
						},

```
and the node labels is something like:

```
{"com.example.csi/zone": "zone10", "com.example.csi/rack": "rack1", "com.example.csi/disk": "ssd"},
```

Note very carefully that the zone value in allowedTopologies is `zone1` and the zone value in the node label is `zone10`. They are different, but the string is a subset of the other.

The first iteration of this PR would have matched these two, which was incorrect. The current implementation correctly does not match them.

Reply:
Yes, good to add that test case.

Reply:
@msau42, @verult, added one negative test case for this.